src/assets/assetsUtil.ts

```diff
@@ -34,3 +38,34 @@ export function compareCollectiblesMetadata(
   }, 0);
   return differentValues > 0;
 }
+
+/**
+ * Compares CollectibleContract entries to any CollectibleContract entry.
+ * We need this method when comparing a new fetched CollectibleContract, in case a entry changed to a defined value,
+ * there's a need to update the CollectibleContract object in state.
+ *
+ * @param newCollectibleContract - CollectibleContract data object.
+ * @param oldCollectibleContract - CollectibleContract object to compare with.
+ * @returns Whether there are differences.
+ */
+export function compareCollectibleContract(
+  newCollectibleContract: CollectibleContract,
+  oldCollectibleContract: CollectibleContract,
+) {
+  const keys: (keyof CollectibleContract)[] = [
```

Comment:
What do you think about moving this outside of the function?

Reply:
done here: https://github.com/MetaMask/controllers/pull/680/commits/2bef09890f6df2392729b8cf4b99623af9322c0f